### PR TITLE
Speed up kaggle-ndsb2 example Preprocessing 

### PR DESCRIPTION
**Pull request by cshaley**
This parallelizes loading, cropping, and resizing images, and it eliminates use of the csvwriter library in that section of code, which seems to be very slow (at least in python 2).  On four cores, this sped up preprocessing time by about 5x - from 45 minutes to 9 minutes for me.  Output is identical to pre-change.  

Added dependencies: 
joblib for parallelization
dill for pickling lambda function

The program no longer prints to screen every hundredth task.

### CODE REVIEW

**Comment by tqchen** [comment]
@antinucleon  can you review and merge this?

**Comment by winstywang** [comment]
Could you check the coding style to make it match to its current form in mxnet? such as spacing, etc.

File: example/kaggle-ndsb2/Preprocessing.py [hunk 1]
```diff
@@ -53,29 +55,33 @@ def write_label_csv(fname, frames, label_map):
...
-       if counter % 100 == 0:
-           print("%d slices processed" % counter)
-   print("All finished, %d slices in total" % counter)
+   dr=Parallel()(delayed(get_data)(lst,preproc) for lst in frames)
+   data,result=zip(*dr)
+   for entry in data:
+      fdata.write(','.join(entry)+'\r\n')
+   print("All finished, %d slices in total" % len(data))
    fdata.close()
+   result=np.ravel(result)
```
Comment (antinucleon) [inline example/kaggle-ndsb2/Preprocessing.py]: style

File: example/kaggle-ndsb2/Preprocessing.py [hunk 2]
```diff
@@ -53,29 +55,33 @@ def write_label_csv(fname, frames, label_map):
...
-           result.append(dst_path)
-           data.append(img)
-       data = np.array(data, dtype=np.uint8)
-       data = data.reshape(data.size)
-       dwriter.writerow(data)
-       counter += 1
-       if counter % 100 == 0:
-           print("%d slices processed" % counter)
-   print("All finished, %d slices in total" % counter)
+   dr=Parallel()(delayed(get_data)(lst,preproc) for lst in frames)
```
Comment (antinucleon) [inline example/kaggle-ndsb2/Preprocessing.py]: style

**Pull request merged**